test/plugin/base_test.rb

```diff
@@ -605,6 +605,66 @@ def test_sanitize_tags_from_container_name_with_require_valid_tags_false
     end
   end
 
+  def test_configure_split_logs_by_tag
+    setup_gce_metadata_stubs
+    {
+      APPLICATION_DEFAULT_CONFIG => true,
+      DISABLE_SPLIT_LOGS_BY_TAG_CONFIG => false
+    }.each do |(config, split_logs_by_tag)|
+      d = create_driver(config)
+      assert_equal split_logs_by_tag,
+                   d.instance.instance_variable_get(:@split_logs_by_tag)
+    end
+  end
+
+  def test_split_logs_by_tag
+    setup_gce_metadata_stubs
+    log_entries_count = 5
+    [
+      [APPLICATION_DEFAULT_CONFIG, log_entries_count, log_entries_count],
+      [DISABLE_SPLIT_LOGS_BY_TAG_CONFIG, 1, log_entries_count]
+    ].each do |(config, successful_requests_count, ingested_entries_count)|
+      setup_prometheus
+      setup_logging_stubs do
+        d = create_driver(config + PROMETHEUS_ENABLE_CONFIG, 'test', true)
+        log_entries_count.times do |i|
+          d.emit("tag#{i}", 'message' => log_entry(0))
+        end
+        d.run
+      end
+      assert_prometheus_metric_value(:stackdriver_successful_requests_count,
+                                     successful_requests_count,
+                                     grpc: use_grpc, code: ok_status_code)
+      assert_prometheus_metric_value(:stackdriver_ingested_entries_count,
+                                     ingested_entries_count,
+                                     grpc: use_grpc, code: ok_status_code)
+    end
+  end
+
+  def test_log_name_when_split_logs_by_tag_disabled
+    setup_gce_metadata_stubs
+    log_entries_count = 5
+    setup_prometheus
+    setup_logging_stubs do
+      d = create_driver(DISABLE_SPLIT_LOGS_BY_TAG_CONFIG +
+                        PROMETHEUS_ENABLE_CONFIG, 'test', true)
+      log_entries_count.times do |i|
+        d.emit("tag#{i}", 'message' => log_entry(0))
+      end
+      d.run
+    end
+    emit_index = 0
+    jsonify_log_entries
+    @logs_sent.each do |request_sent|
+      assert_equal '', request_sent['logName']
```

Comment:
Cool cool. Changed `verify_log_entries` to skip the logName check if params[:log_name] is `nil`. I was debating whether to touch that function at all. But this makes sense i think.

As for the refactoring of `verify_json_log_entries`, yeah, it's not very relevant. I'm slightly leaning towards the refactored version simply because I can easily search for `def verify_lo_entries` in the `BaseTest` file. Most of the time I was debugging in that file. Just a very personal preference though.